File: 13-06-2026/find_missing_integer.py
def find_missing_integer(arr):
    arr = [elem for elem in arr if elem > 0]
    newArr = set(arr)

    if not newArr:
        return 1
    else:
        if len(newArr) == max(newArr):
            return max(newArr) + 1
        else:
            for i in range(1, len(newArr) + 1):
                if i not in newArr:
                    return i

print(find_missing_integer([1, 3, 6, 4, 1, 2])) # 5
print(find_missing_integer([1, 2, 3])) # 4
print(find_missing_integer([-1, -2, 0])) # 1
print(find_missing_integer([2, 3, 7, 6, 8, -1, -10, 15])) # 1

File: 13-06-2026/find_missing_integer.js
function findMissingInteger(A) {
  A = A.filter((elem) => elem > 0).sort((a, b) => a - b);

  if (A.length < 1) return 1;
  else {
    const uniqueSet = [...new Set(A)];
    const [maxValue, minValue] = [
      Math.max(...uniqueSet),
      Math.min(...uniqueSet),
    ];

    if (uniqueSet.length == maxValue) return maxValue + 1;
    else {
      for (let i = 0; i <= maxValue; i++) {
        const [isCurValue, curArr] = [minValue + i, uniqueSet[i]];

        if (isCurValue == curArr) continue;
        return isCurValue;
      }
    }
  }
}

console.log(findMissingInteger([1, 3, 6, 4, 1, 2])); // 5
console.log(findMissingInteger([1, 2, 3])); // 4
console.log(findMissingInteger([-1, -2, 0])); // 1
console.log(findMissingInteger([2, 3, 7, 6, 8, -1, -10, 15]));
